onward/app/services/PopularInTagService.scala

```diff
@@ -0,0 +1,41 @@
+package services
+
+import common.Edition
+import contentapi.ContentApiClient
+import model.RelatedContentItem
+import model.dotcomrendering.{OnwardCollectionResponse, Trail}
+import play.api.mvc.RequestHeader
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PopularInTagService(contentApiClient: ContentApiClient)(implicit
+    executionContext: ExecutionContext,
+) {
+  // `itemViewCounts` is a Map[Content.id: String, ViewCount:Int]
+  // this is generally fetched from Ophan view the MostPopularAgent, but can come from anywhere
+  // and makes this easy to test
+  def fetch(edition: Edition, tag: String, excludeTags: Seq[String], itemViewCounts: Map[String, Int])(implicit
```

Comment:
How do we (cc @AshCorr) feel about calling the new services' `fetch` methods in the existing endpoints as well? e.g. for popular in tag we would call it here: https://github.com/guardian/frontend/blob/main/onward/app/controllers/PopularInTag.scala#L30. 

That would need some refactoring so that `fetch` is able to return both `Future[OnwardCollectionResponse]` (expected by the new endpoint) and `Future[RelatedContent]` (expected by the old endpoint). I paired with Roberto on Friday as I was trying to deal with duplication and we added a method as a parameter to decide the return type: https://github.com/guardian/frontend/blob/290c315b1bd747a09b0d25557c7949d927cbd653/onward/app/services/SeriesService.scala#L15-L18.

I think this is a quite flexible solution and makes the service the source of truth for everything (old endpoints, new endpoints, future agents maybe) but let me know what you both think! Here's the full PR: https://github.com/guardian/frontend/pull/25504/files

Reply:
I'd be wary of coupling the services, part of the reason for this was to make these services DCR specific and easy to change, which we lose by coupling and needing to support things in two places.

We will also be looking to deprecate services that aren't DCR specific one they are no longer used, which would be once this work is done.

Reply:
Makes sense, thanks! Will revert this change in my PR as well